### PR TITLE
Add operation to update OIDC provider configs.

### PR DESCRIPTION
**Pull request by micahstairs**
This adds an operation to update OIDC provider configs (can be done using either the tenant-aware or standard Firebase client).

This work is part of adding multi-tenancy support (see issue #332).

### CODE REVIEW

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java [hunk 1]
```diff
@@ -330,21 +322,45 @@ OidcProviderConfig createOidcProviderConfig(
...
   }
 
   void deleteProviderConfig(String providerId) throws FirebaseAuthException {
-    GenericUrl url = new GenericUrl(idpConfigMgtBaseUrl + "/oauthIdpConfigs/" + providerId);
+    GenericUrl url = new GenericUrl(idpConfigMgtBaseUrl + getOidcUrlSuffix(providerId));
     sendRequest("DELETE", url, null, GenericJson.class);
   }
 
+  private static String generateMask(Map<String, Object> properties) {
+    // This implementation does not currently handle the case of nested properties. This is fine
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/FirebaseUserManager.java:347]: You will need to address this when implementing SAML config support.

File: src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java [hunk 2]
```diff
@@ -71,7 +71,41 @@ public void testCreateRequest() throws IOException {
...
+    Map<String,Object> properties = updateRequest.getProperties();
+    assertEquals(properties.size(), 4);
+    assertEquals("DISPLAY_NAME", (String) properties.get("displayName"));
+    assertFalse((boolean) properties.get("enabled"));
+    assertEquals("CLIENT_ID", (String) properties.get("clientId"));
+    assertEquals("https://oidc.com/issuer", (String) properties.get("issuer"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testUpdateRequestInvalidIssuerUrl() {
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java:108]: Also test for empty/null client ID?

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java [hunk 3]
```diff
@@ -1491,6 +1491,110 @@ public void testTenantAwareCreateOidcProvider() throws Exception {
...
+    OidcProviderConfig config = FirebaseAuth.getInstance().updateOidcProviderConfig(request);
+
+    checkOidcProviderConfig(config);
+    checkRequestHeaders(interceptor);
+    checkUrl(interceptor, "PATCH", PROJECT_BASE_URL + "/oauthIdpConfigs/oidc.provider-id");
+    GenericUrl url = interceptor.getResponse().getRequest().getUrl();
+    assertEquals("displayName", url.getFirst("updateMask"));
+    GenericJson parsed = parseRequestContent(interceptor);
+    assertEquals("DISPLAY_NAME", parsed.get("displayName"));
+    assertNull(parsed.get("enabled"));
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java:1535]:
```
assertEquals(1, parsed.size());
```

File: src/main/java/com/google/firebase/auth/ProviderConfig.java [hunk 4]
```diff
@@ -104,4 +104,48 @@ public T setEnabled(boolean enabled) {
...
+  /**
+   * A base class for updating the attributes of an existing provider.
+   */
+  public abstract static class AbstractUpdateRequest<T extends AbstractUpdateRequest<T>> {
+
+    final String providerId;
+    final Map<String,Object> properties = new HashMap<>();
+
+    AbstractUpdateRequest(String providerId) {
+      this.providerId = providerId;
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/ProviderConfig.java:117]: check not null or empty

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java (hunk 1 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/FirebaseUserManager.java:347]: Yeah I'll defer it until then.

File: src/main/java/com/google/firebase/auth/ProviderConfig.java (hunk 4 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/ProviderConfig.java:117]: I already had the check in OidcProviderConfig (so that it could be closer to the Javadoc comment which mentions that it throws an IllegalArgumentException if it's null or empty. But I moved it up here.

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java (hunk 3 above)
Comment (micahstairs) [reply src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java:1535]: Done.

File: src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java (hunk 2 above)
Comment (micahstairs) [reply src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java:108]: Done.

Comment (hiranya911) [reply src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java:113]: I was particularly referring to empty/null client ID: `.setClientId(null)`. Not sure if the current tests cover that case. But feel free to address it in a future PR.

Comment (micahstairs) [reply src/test/java/com/google/firebase/auth/OidcProviderConfigTest.java:123]: Ah, that's easy enough to add here. Done.
